stl/inc/ranges

```diff
@@ -1433,6 +1463,94 @@ namespace ranges {
         inline constexpr _Istream_fn<_Ty> istream;
     } // namespace views
 
+#if _HAS_CXX23
+    namespace _Pipe {
+        template <class _Ty>
+        void _Derived_from_range_adaptor_closure_impl(const volatile range_adaptor_closure<_Ty>&);
+
+        template <class _Ty>
+        concept _Derived_from_range_adaptor_closure = requires(const volatile _Ty& __obj) {
+            _Derived_from_range_adaptor_closure_impl(__obj);
+        };
+
+        template <class _Ty>
+        concept _Is_range_adaptor_common =
+            derived_from<remove_cvref_t<_Ty>, range_adaptor_closure<remove_cvref_t<_Ty>>> //

```

Comment:
The latter `remove_cvref_t` may not be necessary since `_Ty` must be a class type (i.e., not a reference), and must not be cv-qualified according to the requirements on `range_adaptor_closure`.